src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -61355,6 +61355,281 @@ void F(C x)
                 );
         }
 
+        // https://github.com/dotnet/csharplang/blob/master/meetings/2019/LDM-2019-12-18.md
+        [Fact]
+        public void OtherComparisonsAsPureNullTests_01()
+        {
+            var source =
+@"#nullable enable
+using System;
+class E { }
+class Program
+{
+    static void F1(E e)
+    {
+        if (e is object) return;
+        e.ToString(); // 1
+    }
+    static void F2(E e)
+    {
+        if (e is Object) return;
```

Comment:
>Object [](start = 17, length = 6)

Please add a test for `e is SomeName` where `SomeName` is a using aliad for `System.Object`.
Please add a test for `e is dynamic`.  Because it is "equivalent" to `object`, I believe that should work as a pure test.

Reply:
Added. 

`e is dynamic` is not treated as a pure null test.

---
In reply to: [366014818](https://github.com/dotnet/roslyn/pull/40892#discussion_r366014818) [](ancestors = 366014818)